trieste/acquisition/rule.py

```diff
@@ -290,7 +293,75 @@ def acquire(
             )
 
         acquisition_function = self._builder.prepare_acquisition_function(datasets, models)
+
         point = _optimizer.optimize(acquisition_space, acquisition_function)
         state_ = TrustRegion.State(acquisition_space, eps, y_min, is_global)
 
         return point, state_
+
+
+class BatchAcquisitionRule(AcquisitionRule[None, SearchSpace]):
+    """ Implements a acquisition rule for a batch of query points """
+
+    def __init__(self, num_query_points: int,
+                 builder: Optional[AcquisitionFunctionBuilder] = None):
+        """
+        :param num_query_points: The number of points to acquire.
+        :param builder: The acquisition function builder to use.
+            :class:`EfficientGlobalOptimization` will attempt to **maximise** the corresponding
+            acquisition function. Defaults to :class:`~trieste.acquisition.ExpectedImprovement`
+            with tag `OBJECTIVE`.
+        """
+
+        if not num_query_points > 0:
+            raise ValueError(
+                f"Number of query points must be greater than 0, got {num_query_points}"
+            )
+
+        self._num_query_points = num_query_points
+
+        if builder is None:
+            num_samples = 100
+            builder = MonteCarloExpectedImprovement(eps_shape=[num_samples, num_query_points, 1]).using(OBJECTIVE)  # [S, B, L]
+
+        self._builder = builder
+
+    def expand_search_space(self, search_space: SearchSpace):
+        assert isinstance(search_space, Box)
+
+        lower_bound, upper_bound = search_space.lower, search_space.upper
+        expanded_lower_bound = tf.tile(lower_bound, [self._num_query_points])
+        expanded_upper_bound = tf.tile(upper_bound, [self._num_query_points])
+        return Box(expanded_lower_bound, expanded_upper_bound)
```

Comment:
any reason we can't do this for discrete search spaces?

Reply:
e.g. can we tile all the points in a discrete search space and achieve the same?

Reply:
no reason why not. But I think that this `expand` should be a feature of the search space. Should we create a PR for that?